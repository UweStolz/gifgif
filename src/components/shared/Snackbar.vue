<template>
  <v-snackbar
    :value="syncVisibility"
    :timeout="timeout"
    :color="color"
    top
  >
    {{ message }}
    <v-btn
      v-if="showCloseButton"
      dark
      text
      icon
      @click="syncVisibility = false"
    >
      <v-icon>
        {{ icons.mdiClose }}<v-icon />
      </v-icon>
    </v-btn>
  </v-snackbar>
</template>

<script lang="ts">
import {
  Component, Vue, Prop, PropSync,
} from 'vue-property-decorator';
import { mdiClose } from '@mdi/js';

@Component
export default class Snackbar extends Vue {
@PropSync('visibility', { default: false }) syncVisibility!: boolean

@Prop({ default: 'info' }) color!: string

@Prop({ default: false }) showCloseButton!: boolean;

@Prop({ default: 'Whoops something went wrong' }) message!: string

@Prop({ default: 3000 }) timeout: number = 3000;

icons = { mdiClose }
}
</script>
