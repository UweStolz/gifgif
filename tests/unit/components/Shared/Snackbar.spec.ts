import Snackbar from '@/components/shared/Snackbar.vue';
import shallow from '../../../helper';

describe('Snackbar.vue', () => {
  it('renders properly', () => {
    const wrapper = shallow(Snackbar);
    expect(wrapper.element).toMatchSnapshot();
  });
});
