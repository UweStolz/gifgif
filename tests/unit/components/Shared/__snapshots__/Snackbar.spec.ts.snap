// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar.vue renders properly 1`] = `
<v-snackbar-stub
  color="info"
  timeout="3000"
  top="true"
>
  <!---->
  
  Whoops something went wrong
  
  <!---->
</v-snackbar-stub>
`;
